src/components/verification-grid/verification-grid.ts

```diff
@@ -933,6 +1054,158 @@ export class VerificationGridComponent extends WithShoelace(AbstractComponent(Li
+    }
+
+    const selectionIndex = tileIndices;
+
+    // in desktop mode, unless the ctrl key is held down, clicking an element
+    // removes all other selected items
+    // while it is not possible to press the ctrl key on a tablet, the user can
+    // still overwrite the selection behavior using the selection-behavior
+    // attribute. Therefore, we have to check that we are not on explicitly
+    // using tablet selection mode.
+    if (selectionBehavior === "desktop" && !toggle && !additive && !focus) {
+      this.removeSubSelection();
+    }
+
+    const iterableIndices = Array.isArray(selectionIndex) ? selectionIndex : [selectionIndex];
+    for (const tileIndex of iterableIndices) {
+      // If the "focus" selection behavior is set, we want to focus the tile but
+      // not select it.
+      if (focus) {
+        this.focusTile(tileIndex);
+        this.focusHead = tileIndex;
+      } else if (range) {
+        // if the user has never selected an item before, the multiSelectHead will be "null"
+        // in this case, we want to start selecting from the clicked tile
+        this.rangeSelectionHead ??= tileIndex;
+        const selectionTail = tileIndex;
+
+        this.addSubSelectionRange(this.rangeSelectionHead, selectionTail);
+      } else if (additive) {
+        this.selectTile(tileIndex);
+        this.rangeSelectionHead = tileIndex;
+      } else {
+        // if we reach this point, we know that the user is not performing a
+        // range selection because range selection performs an early return
+        this.toggleTileSelection(tileIndex);
+        this.rangeSelectionHead = tileIndex;
+      }
+    }
+
+    this.updateSubSelection();
+  }
+
+  private resetSelectionHead(): void {
+    this.rangeSelectionHead = null;
+  }
+
+  private updateSelectionHead(value: number | null, options?: SelectionOptions): void {
+    const refinedOptions: SelectionOptions = {
+      focus: options?.toggle,
+      range: options?.range,
+    };
+
+    if (value !== null) {
+      this.processSelection(value, refinedOptions);
+    } else {
+      this.removeSubSelection();
+    }
+  }
+
+  private selectionHeadUndecided(selectionIndex: number): void {
+    let firstUndecidedTile: number | null = null;
+    const nextUndecidedTile = Array.from(this.gridTiles).findIndex((tile, index) => {
+      // We include the current index in the undecided tile check, so that if
+      // the current tile that the user is on is still undenied, we will stay
+      // on the current tile.
+      // Note that you should not be calling this function if the current tile
+      // is incomplete, but this is a defensive programming / double check to
+      // ensure that we don't skip over the currently incomplete tile.
+      if (index < selectionIndex) {
+        if (firstUndecidedTile === null && !tile.taskCompleted) {
+          firstUndecidedTile = index;
+        }
+
+        return false;
+      }
+
+      return !tile.taskCompleted;
+    });
```

Comment:
i this case i'd probably favour literate code over performant code:

```
fistUndecidedTile = this.getFirstUndecidedTile();
nextUndecidedTile = this.getNextUndecidedTile();
```

Reply:
I agree on that philosophy

However, you'll have to explain why this is unreadable, (as the author), this seems very flat and and simple to me.